nwbwidgets/misc.py

```diff
@@ -1087,3 +1092,196 @@ def show_session_raster_plotly(
+        if trials is None:
+            self.trials = self.get_trials()
+            if self.trials is None:
+                self.children = [widgets.HTML("No trials present")]
+                return
+        else:
+            self.trials = trials
+
+        # Create variables choice dropdowns
+        groups = self.get_groups(self.trials)
+        self.rows_controller = widgets.Dropdown(
+            options=[None] + list(groups), 
+            description="rows",
+            value=None
+        )
+        self.rows_controller.observe(self.rows_callback, names='value')
+
+        self.cols_controller = widgets.Dropdown(
+            options=[None] + list(groups), 
+            description="cols",
+            disabled=True,
+        )
+
+        # Unit controller
+        unit_ids = units.id.data[:]
+        n_units = len(unit_ids)
+        self.unit_controller = widgets.Dropdown(
+            options=[(str(unit_ids[x]), x) for x in range(n_units)],
+            value=unit_index,
+            description="unit",
+        )
+
+        # Trial event controller (align by) 
+        self.trial_event_controller = make_trial_event_controller(self.trials)
+
+        # Before / After controllers
+        before_slider = widgets.FloatSlider(
+            0.1, min=0, max=5.0, description="before (s)", continuous_update=False
+        )
+        after_slider = widgets.FloatSlider(
+            1.0, min=0, max=5.0, description="after (s)", continuous_update=False
+        )
+
+        self.controls = {
+            "units": fixed(units),
+            "time_intervals": fixed(self.trials),
+            "index": self.unit_controller,
+            "after": after_slider,
+            "before": before_slider,
+            "align_by": self.trial_event_controller,
+            "rows_label": self.rows_controller,
+            "cols_label": self.cols_controller,
+        }
+
+        self.fig_tuning_curve = interactive_output(draw_tuning_curve, self.controls)
+        self.fig_raster_grid = interactive_output(raster_grid, self.controls)
+        
+        self.children = [
+            self.unit_controller,
+            self.rows_controller,
+            self.cols_controller,
+            self.trial_event_controller,
+            before_slider,
+            after_slider,
+            self.fig_tuning_curve,
+            self.fig_raster_grid
+        ]
+
+    def get_trials(self):
+        return self.units.get_ancestor("NWBFile").trials
+
+    def get_groups(self, trials):
+        return infer_categorical_columns(dynamic_table=trials)
+
+    def rows_callback(self, change):
+        """
+        Gets triggered when self.rows_controller changes. Updates other dropdown options.
+        """
+        if change['new'] is None:
+            self.cols_controller.disabled = True
+            self.cols_controller.value = None
+        else:
+            self.cols_controller.disabled = False
+
+
+def draw_tuning_curve(
```

Comment:
Could you separate this out into 2 functions, routed by whether col_label is None?

Reply:
done